src/slice/mod.rs

```diff
@@ -775,6 +890,61 @@ impl<'data, T: Send + 'data> IndexedParallelIterator for ChunksMut<'data, T> {
     }
 }
 
+/// Parallel iterator over mutable non-overlapping chunks of a slice
+#[derive(Debug)]
+pub struct ChunksExactMut<'data, T: 'data + Send> {
+    chunk_size: usize,
+    slice: &'data mut [T],
+    rem: &'data mut [T],
+}
+
+impl<'data, T: Send> ChunksExactMut<'data, T> {
+    /// Return the remainder of the original slice that is not going to be
+    /// returned by the iterator. The returned slice has at most `chunk_size-1`
+    /// elements.
+    pub fn into_remainder(self) -> &'data mut [T] {
```

Comment:
This is awkward to consume `self`, because it means the rest of the `ParallelIterator` can't be used at all. With `Iterator` you can at least pull the items out first without consuming `self`, with `next`, `by_ref`, etc., but not so in rayon. All of the parallel methods consume `self`, so you can't do that *and* get the remainder here.

Maybe we can keep `into_remainder` for `std` parity, but also add something non-consuming like:

```rust
    pub fn remainder(&mut self) -> &mut [T] {
        self.rem                // ^ lifetime borrowed from self, not 'data
    }

    pub fn take_remainder(&mut self) -> &'data mut [T] {
        mem::replace(&mut self.rem, &mut [])
    }
```